File: SILVER/S4_28278_스택 2.py
def push(x):
    stack.append(x)

def pop():
    if stack:
        print(stack.pop(-1))
    else:
        print(-1)

def size():
    print(len(stack))

def empty():
    if not stack:
        print(1)
    else:
        print(0)

def top():
    if stack:
        print(stack[-1])
    else:
        print(-1)

import sys
input = sys.stdin.readline

n = int(input())
stack = []

for _ in range(n):
    command = list(map(int, input().split()))
    if command[0] == 1:
        push(command[1])
    elif command[0] == 2:
        pop()
    elif command[0] == 3:
        size()
    elif command[0] == 4:
        empty()
    elif command[0] == 5:
        top()

File: SILVER/S4_28278_스택 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        StringTokenizer st;
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            String s = br.readLine();
            st = new StringTokenizer(s);

            int command = Integer.parseInt(st.nextToken());

            switch (command) {
                // 스택에 숫자 삽입
                case 1:
                    int num = Integer.parseInt(st.nextToken());
                    stack.push(num);
                    break;

                // 스택의 가장 위에 있는 정수를 pop하고 출력
                case 2:
                    // 비어있다면 -1 출력
                    if (stack.empty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(stack.pop()).append("\n");
                    }
                    break;

                // 스택에 들어있는 정수 개수
                case 3:
                    sb.append(stack.size()).append("\n");
                    break;

                // 스택이 비어있다면 1, 아니면 0 출력
                case 4:
                    if (stack.empty()) {
                        sb.append(1).append("\n");
                    } else {
                        sb.append(0).append("\n");
                    }
                    break;

                // 스택의 가장 위에 있는 정수를 pop하지 않고 출력
                case 5:
                    // 비어있다면 -1 출력
                    if (stack.empty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(stack.peek()).append("\n");
                    }
                    break;
            }
        }

        System.out.println(sb);
    }
}